luaui/Widgets/cmd_blueprint.lua

```diff
@@ -411,9 +477,16 @@ local function deleteBlueprint(index)
 
 	if #blueprints == 0 then
 		setSelectedBlueprintIndex(nil)
-	elseif selectedBlueprintIndex > #blueprints then
-		setSelectedBlueprintIndex(selectedBlueprintIndex - 1)
+	elseif index > selectedBlueprintIndex then
+		-- no need to do anything
+	elseif index == selectedBlueprintIndex then
+		-- find the closest valid blueprint, searching backwards
+		setSelectedBlueprintIndex(
+			getPrevFilteredBlueprintIndex(selectedBlueprintIndex)
+		)
+		lastExplicitlySelectedBlueprintIndex = selectedBlueprintIndex
 	elseif index < selectedBlueprintIndex then
+		-- keep the same blueprint selected
 		setSelectedBlueprintIndex(selectedBlueprintIndex - 1)
```

Comment:
not sure if you keeping this (because of previous comment on this method), but if you do, maybe you should set lastExplicitlySelectedBlueprintIndex here too? For consistency with the previous case looks like you should.